Applications/Utils/ModelPreparation/PartitionMesh/MeshPartitioning.cpp

```diff
@@ -0,0 +1,493 @@
+/*!
+  \file MeshPartitioning.cpp
+  \date   2016.05
+
+  \brief  Define the members of class MeshPartitioning
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#include "MeshPartitioning.h"
+
+#include <iomanip>
+#include <stdio.h> // for binary output
+
+#include "logog/include/logog.hpp"
+
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Element.h"
+
+namespace MeshLib
+{
+void MeshPartitioning :: write2METIS(const std::string& file_name)
+{
+    std::ofstream os(file_name.data(), std::ios::trunc);
```

Comment:
Maybe we should introduce checked file opening routines in a subsequent PR and then use those in all of our code, because I think there are similar issues in several other places.